plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Products.php

```diff
@@ -223,7 +244,9 @@ private static function count_user_products() {
 
 		$products_query = new \WP_Query( $args );
 
-		return $products_query->found_posts;
+		$value = $products_query->post_count > 0 ? 'yes' : 'no';
```

Comment:
I could be misreading the code but I think if you change this to only store `$products_query->post_count > 0` instead of the string. Then up on L222 you can check for `$product_exists !== false` which could save a DB query in the event get_transient is returning false due to being set to false and not simply missing.

Reply:
I believe that might actually be less efficient and never allow the negative case to use the transient.  For example:

```php
$product_exists = get_transient( self::HAS_PRODUCT_TRANSIENT );
if ( $product_exists !== false ) {
    // We won't ever come in here if the transient was set to false, so if the query has previously run, the transient can't be used and this forces us to re-query below.
    return $product_exists;
}
```

Let me know if I misunderstood your suggestion!

Reply:
Yeah you're right, the storing of strings for future eval was a bit of a code smell for me and I thought I could save a db query in the process but yeah we'd end up not hitting it in some cases making it worse 👍